Objective-C/CBLArray.m

```diff
@@ -0,0 +1,343 @@
+//
+//  CBLArraym
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 4/12/17.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+#import "CBLArray.h"
+#import "CBLBlob.h"
+#import "CBLData.h"
+#import "CBLDocument+Internal.h"
+#import "CBLJSON.h"
+#import "CBLSubdocument.h"
+
+
+@implementation CBLArray {
```

Comment:
I don't thing we have ever thought about the sibling relationship. I created https://github.com/couchbaselabs/couchbase-lite-apiv2/issues/84 for this.